### ISSUE TITLE
`typing.Optional` type is being confused with `pyparsing.core.Optional`

### ISSUE THREAD

**Issue by jtprobst** [python/mypy#12563]
**Bug Report**

Apparently, `mypy` may fail when the name `Optional` is used in libraries in certain situations. [Please see also this corresponding bug report in pyparsing.](https://github.com/pyparsing/pyparsing/issues/385#issuecomment-1094913196) and a [duplicate thereof.](https://github.com/pyparsing/pyparsing/issues/387)

**To Reproduce**

(Write your steps here:)

1. Install `mypy==0.942`, `mypy-extensions==0.4.3` and `pyparsing==3.0.8` in a venv with (Python 3.8.13)
2. Write the following to `scratch.py`:
```
from pyparsing import Literal, Optional

Literal("(") + Optional(Literal(")"))
```
3. Typecheck with `$ mypy scratch.py`

**Expected Behavior**

No type errors.

**Actual Behavior**

Type checking fails with the following error:

```
scratch.py:3: error: "_SpecialForm" not callable
Found 1 error in 1 file (checked 1 source file)
```

**Your Environment**

- Mypy version used: 0.942
- Mypy command-line flags: none
- Mypy configuration options from `mypy.ini` (and other config files): none
- Python version used: 3.8.13
- Operating system and version: Ubuntu 18.04

**Comment by erictraut** [comment]
It appears this bug has been fixed. The spurious error no longer occurs with mypy 1.5.
